internal/cache/cache.go

```diff
@@ -0,0 +1,37 @@
+package cache
+
+import (
+	"errors"
+	"time"
+)
+
+// ExpiryDuration is used to define the expiry duration when caching some data.
+type ExpiryDuration uint8
```

Comment:
why using uint?

Reply:
do not get confused with this usage, it just to make sure we only use positive value for `ExpiryConf` key. 
we can use `string`:
```go
expirationTime = map[string]time.Duration{
	"DurationShort": 5 * time.Minute,
	"DurationLong":  10 * time.Minute,
}
```
we can use `int`:
```go
expirationTime = map[string]time.Duration{
	1: 5 * time.Minute,
	-2:  10 * time.Minute,
}
```
but for this case I just to make sure we only use positive int value:
```go
expirationTime = map[string]time.Duration{
	1: 5 * time.Minute,
	2:  10 * time.Minute,
}
```